python/cudf/cudf/tests/test_index.py

```diff
@@ -2316,3 +2316,22 @@ def test_get_loc_multi_string(idx, key, method):
         got = gi.get_loc(key, method=method)
 
         assert_eq(expected, got)
+
+
+@pytest.mark.parametrize(
+    "objs",
+    [
+        [pd.RangeIndex(0, 10), pd.RangeIndex(10, 20)],
+        [pd.RangeIndex(10, 20), pd.RangeIndex(22, 40), pd.RangeIndex(50, 60)],
+        [pd.RangeIndex(10, 20, 2), pd.RangeIndex(20, 40, 2)],
+    ],
+)
+def test_range_index_concat(objs):
+    cudf_objs = [cudf.from_pandas(obj) for obj in objs]
+
+    actual = cudf.concat(cudf_objs)
+
+    expected = objs[0]
+    for obj in objs[1:]:
+        expected = expected.append(obj)
```

Comment:
This test will fail because `expected` becomes `None` after the first iteration because `append` does not return the input. However, I'm also not entirely sure what output we're expecting here, should we get a list of `RangeIndex` objects? I think `extend` is what we want, not `append`, right? If so, perhaps this would be a simpler solution?
```suggestion
    expected = sum(obs, [])
```

Reply:
`expected` is a `RangeIndex` here, and `RangeIndex.append` does return a resulting Index.

Reply:
Ah I see sorry I misread the parametrization and thought that `objs` was actually coming in as a list of list of `RangeIndex` objects rather than just a flat list of `RangeIndex` across multiple iterations. OK then I think that this is good to go.